packages/app/src/state/staking/reducer.ts

```diff
@@ -10,31 +10,43 @@ import {
 	fetchClaimableRewards,
 	fetchEscrowData,
 	fetchStakingData,
-	getReward,
+	claimStakingRewards,
 	stakeEscrow,
 	stakeKwenta,
 	unstakeEscrow,
 	unstakeKwenta,
 	vestEscrowedRewards,
+	fetchStakingV2Data,
+	fetchEscrowV2Data,
+	fetchStakeMigrateData,
 } from './actions'
 import { StakingState } from './types'
 
 export const STAKING_INITIAL_STATE: StakingState = {
 	kwentaBalance: '0',
 	escrowedKwentaBalance: '0',
+	escrowedKwentaBalanceV2: '0',
```

Comment:
as this root object is growing quite big it might be nice to break it up a bit, potentially with a `v1` and `v2` field. Something like:

```
VersionedStakeData = {
  claimableBalance: string
  totalStakedBalance: string
  stakedEscrowedKwentaBalance: string
  escrowedKwentaBalance: string
}

StakingState = {
  ...
  v1: VersionedStakeData
  v2: VersionedStakeData
}
```

Reply:
Good idea. Will follow your advice.